src/tools/tablegen_rewrites_to_elevate.py

```diff
@@ -0,0 +1,395 @@
+from __future__ import annotations
+from dataclasses import dataclass, field
+from enum import Enum
+from io import StringIO, TextIOWrapper
+import re
+import sys
+from typing import List, Optional, Sequence, Type
+
+from xdsl.dialects.onnx import *
+
+from tablegen_dialect_def_to_irdl import *
+from xdsl.ir import Operation
+
+
+class AdvanceMode(Enum):
+    SkipUntilMatch = True
+    NoSkip = False
+
+
+@dataclass
+class Value():
+    name: str
+
+    def get_matching_code(self, dialect_name: str) -> str:
+        return self.name
+
+    def get_replacement_code(self, dialect_name: str) -> str:
+        return self.name
+
+
+@dataclass
+class ParsedAttribute(Value):
+    name_def: str
+
+    def get_matching_code(self, dialect_name: str) -> str:
+        return f"\"{self.name_def}\" : {self.name}"
+
+
+@dataclass
+class NativeCodeCall():
+    name: str
+
+    def get_matching_code(self, dialect_name: str) -> str:
+        return self.name + "()"
+
+    def get_replacement_code(self, dialect_name: str) -> str:
+        return self.name + "()"
+
+
+@dataclass
+class Op():
+    name: str
+    type: Type[Operation]
+    operands: List[Value | Op]
+    attributes: List[Value | ParsedAttribute | NativeCodeCall]
+    result_types: List[Attribute]
```

Comment:
```suggestion
    result_types: list[Attribute]
```
Same for above three lines as typing.List has been deprecated since python 3.9